### PR TITLE
Add SCSS support

### PR DESCRIPTION
**Pull request by ducksoupdev**
I use this atom plugin all the time for css and it's awesome so I thought I'd add the postcss-scss parser for working with scss. 

Let me know if there is anything I need to change.

### CODE REVIEW

**Comment by sindresorhus** [comment]
You can't just replace the existing parser, right? Scss has different semantics than CSS.

**Comment by ducksoupdev** [comment]
I agree with you but it looks like it handles CSS as well. I ran a number of manual tests against different  CSS and SCSS files and the parser works!

File: .jshintrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+	"esversion": 6
+}
```
Comment (sindresorhus) [inline .jshintrc]: I don't use JSHint.

**Comment by sindresorhus** [comment]
Even though it looks like it works perfectly, it might not always. I would prefer to only use the SCSS parser on SCSS syntax.

@ai Any opinion on this?

**Comment by ai** [comment]
@sindresorhus postcss-scss is pretty stable. Sass team looked into it and didn't find big mistakes on first look.

Anyway it is better, than CSS default parser for SCSS sources.

File: .jshintrc (hunk 1 above)
Comment (ducksoupdev) [inline .jshintrc]: I'll remove that.

**Comment by ducksoupdev** [comment]
I agree with you @sindresorhus on using the correct parser for the syntax. However, syntax detection would be difficult when the editor has a buffer that isn't saved. Any ideas?

**Comment by sindresorhus** [comment]
> However, syntax detection would be difficult when the editor has a buffer that isn't saved. Any ideas?

Then use the SCSS parser in those cases. That way it will always work.

**Comment by ducksoupdev** [comment]
OK so use the safe parser if the file is a CSS file otherwise use the SCSS parser?

**Comment by sindresorhus** [comment]
Yes

**Comment by ducksoupdev** [comment]
I realise there are now more commits! Am happy to squash and re-submit the PR?

File: index.js [hunk 2]
```diff
@@ -12,13 +21,22 @@ function init() {
 
 	const selectedText = editor.getSelectedText();
 	const text = selectedText || editor.getText();
+	const path = editor.getPath();
+	let postcssParser = postcssScssParser;
+
+	if (path && ext(path) === 'css') {
```
Comment (sindresorhus) [inline index.js]: just use `require('path').extname().toLowerCase()`

File: index.js [hunk 3]
```diff
@@ -1,7 +1,16 @@
+
```
Comment (sindresorhus) [inline index.js]: no

File: index.js [hunk 4]
```diff
@@ -12,13 +21,22 @@ function init() {
 
 	const selectedText = editor.getSelectedText();
 	const text = selectedText || editor.getText();
+	const path = editor.getPath();
+	let postcssParser = postcssScssParser;
+
+	if (path && ext(path) === 'css') {
+		console.info('Using safe css parser');
```
Comment (sindresorhus) [inline index.js]: no

**Comment by sindresorhus** [comment]
Thank you :)

**Comment by janscarton** [comment]
There needs to be a config flag to disable this. I want vendor prefixes in my compiled CSS, but NOT in my SCSS!

**Pull request closed without merging**